benchmarks/set_http_meta/scenario.py

```diff
@@ -0,0 +1,130 @@
+    "HTTP_SEC_FETCH_SITE": "none",
+    "HTTP_SEC_FETCH_MODE": "navigate",
+    "HTTP_SEC_FETCH_USER": "?1",
+    "HTTP_SEC_FETCH_DEST": "document",
+    "HTTP_ACCEPT_ENCODING": "gzip, deflate, br",
+    "HTTP_ACCEPT_LANGUAGE": "en-US,en;q=0.9",
+    "HTTP_COOKIE": "Pycharm-45729245=449f1b16-fe0a-4623-92bc-418ec418ed4b; Idea-9fdb9ed8="
+    "448d4c93-863c-4e9b-a8e7-bbfbacd073d2; csrftoken=cR8TVoVebF2afssCR16pQeqHcxA"
+    "lA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL; _xsrf=2|d4b85683|7e2604058ea673d12dc6604f"
+    '96e6e06d|1635869800; username-localhost-8888="2|1:0|10:1637328584|23:username-loca'
+    "lhost-8888|44:OWNiOTFhMjg1NDllNDQxY2I2Y2M2ODViMzRjMTg3NGU=|3bc68f938dcc081a9a02e51660"
+    '0c0d38b14a3032053a7e16b180839298e25b42"',
+    "wsgi.input": BytesIO(),
+    "wsgi.url_scheme": "http",
+}
+
+COOKIES = {"csrftoken": "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL"}
+
+DATA_GET = dict(
+    method="GET",
+    url="http://localhost:8888{}".format(PATH),
+    status_code=200,
+    status_msg="OK",
+    query=b"key1=value1&key2=value2&token=cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL",
+    parsed_query={
+        "key1": "value1",
+        "key2": "value2",
+        "token": "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL",
+    },
+    request_headers=COMMON_DJANGO_META,
+    response_headers=COMMON_DJANGO_META,
+    retries_remain=0,
+    raw_uri="http://localhost:8888{}?key1=value1&key2=value2&token="
+    "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL".format(PATH),
+    request_cookies=COOKIES,
+    request_path_params={"id": 1},
+)
+
+DATA_POST = dict(
+    method="GET",
+    url="http://localhost:8888/test-benchmark",
+    status_code=200,
+    status_msg="OK",
+    query=b"key1=value1&key2=value2",
+    parsed_query={"key1": "value1", "key2": "value2"},
+    request_headers=COMMON_DJANGO_META,
+    response_headers=COMMON_DJANGO_META,
+    retries_remain=0,
+    raw_uri="http://127.0.0.1:8081/test/1/?key1=value1&key2=value2",
+    request_cookies=COOKIES,
+    request_path_params={"id": 1},
+    request_body=None,
+)
+
+
+class SetHttpMeta(bm.Scenario):
+    allenabled = bm.var_bool()
+    useragentvariant = bm.var(type=str)
+
+    def run(self):
+        # run scenario to also set tags on spans
+        if self.allenabled:
+            config = Config(lambda: True)
+        else:
+            config = Config(lambda: False)
+
+        data = copy.deepcopy(DATA_GET)
+
+        if self.useragentvariant:
+            data["request_headers"][self.useragentvariant] = (
+                "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 "
+                "(KHTML, like Gecko) Chrome/92.0.4515.159 Safari/537.36"
+            )
+
+        span = utils.gen_span(str("test"))
+        span._local_root = utils.gen_span(str("root"))
+
+        def bm(loops):
+            for _ in range(loops):
+                for _ in range(10000):
```

Comment:
Can you explain why you do this 10k times within the inner loop? This might be better pulled out as a parameter for the scenario.

More generally, do we expect the timing to be the same the first time this function is called as it is for the last time it is called?

Reply:
Ah was it a matter of having nanosecond timings that had random variations? I wonder if this has to do with how it can be hard to get good results when running locally during development.